src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/ViewExecutor.cs

```diff
@@ -84,12 +122,27 @@ public static class ViewExecutor
                     viewData,
                     tempData,
                     writer,
-                    htmlHelperOptions);
+                    ViewOptions.HtmlHelperOptions);
+
+                if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.BeforeView"))
+                {
+                    Telemetry.WriteTelemetry(
+                        "Microsoft.AspNet.Mvc.BeforeView",
+                        new { view = view, viewContext = viewContext, });
+                }
 
                 await view.RenderAsync(viewContext);
-                // Invoke FlushAsync to ensure any buffered content is asynchronously written to the underlying
-                // response. In the absence of this line, the buffer gets synchronously written to the response
-                // as part of the Dispose which has a perf impact.
+
+                if (Telemetry.IsEnabled("Microsoft.AspNet.Mvc.AfterView"))
+                {
+                    Telemetry.WriteTelemetry(
+                        "Microsoft.AspNet.Mvc.AfterView",
+                        new { view = view, viewContext = viewContext, });
+                }
+
```

Comment:
> We can also include the actionresult here if you want.

I deleted this comment because it's not that simple. If the actionID is what you want, you have that the `ViewContext` here already.

> we show them differently in the ui.

This is for top-level results **only** not for `@Html.Partial` - `RazorView` actually already knows if it's a partial or not.